### PR TITLE
Development update

### PR DESCRIPTION
**Pull request by turcato-niccolo**
## Premise

PR request opened follwing requests from: https://trello.com/c/BAILPUgt/65-situation-with-the-find-my-phone-app

## Changes

- Updated the AlarmManager and AlarmMessageHelper to support response messages

- SMSContactManager converted to singleton

### Footer

Check out my Tree [Carlo!](https://www.treedom.net/it/user/niccolo-turcato/trees/YPR-8E3?filter=all)
"BARBASCURA15" for a 15% discount if you want to buy one too

### CODE REVIEW

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5060801&bid=16655812)

File: app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java [hunk 1]
```diff
@@ -33,14 +31,28 @@
...
-    public SMSContactManager(Context applicationContext) {
+    private SMSContactManager(Context applicationContext) {
         contactDatabase = Room.databaseBuilder(applicationContext, SMSContactDatabase.class,
                 CONTACTS_DB_NAME)
                 .enableMultiInstanceInvalidation()
                 .allowMainThreadQueries()
                 .build();
     }
 
+    /**
```
Comment (bortoletti-giorgia) [inline app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:42]: Maybe I missed the message where you said you did the Singleton but I saw it free on Trello and I did it too. I marked it on trello some hours ago. It is in the related contact branch.
Tomorrow I will open a pull request with the branch `feature/contacts`.

File: app/src/test/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelperTest.java [hunk 2]
```diff
@@ -14,11 +14,17 @@
 
     private String alarmMessageRequest;
     private String notAlarmMessageRequest;
+    private String alarmMessageResponse;
+    private String notAlarmMessageResponse;
+    private double time;
```
Comment (bortoletti-giorgia) [inline app/src/test/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelperTest.java]: I suggest you to make final `time `if you don't change it.

File: app/src/test/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelperTest.java [hunk 3]
```diff
@@ -14,11 +14,17 @@
 
     private String alarmMessageRequest;
     private String notAlarmMessageRequest;
+    private String alarmMessageResponse;
```
Comment (bortoletti-giorgia) [inline app/src/test/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelperTest.java:17]: You can add your name in the authors.

File: app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java [hunk 4]
```diff
@@ -23,8 +23,8 @@
     /**
      * Verify if the text receive contains the default audioAlarmMessageRequest sets in this class
      *
-     * @param messageReceived the text message received
-     * @return true if the received text contains the (formatted) alarm Request
+     * @param messageReceived   The received text message
```
Comment (bortoletti-giorgia) [inline app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java:26]: Add your name in the authors.

File: app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java (hunk 1 above)
Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java:42]: That's strange, I saw it free and before doing anything I marked it done xD

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5060801&bid=16655812)

File: .idea/misc.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
     <option name="OUTPUT_DIRECTORY" value="$PROJECT_DIR$/javadoc" />
     <option name="OPTION_SCOPE" value="private" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (ricdezen) [inline .idea/misc.xml]: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa. I don't even know what causes this anymore lmao.

File: app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java [hunk 6]
```diff
@@ -39,6 +39,23 @@ public String getAlarmRequestMessage() {
         return AlarmMessageHelper.composeRequestAlarm();
     }
 
+    /**
+     * Get an alarm response message composed by {@link AlarmMessageHelper}
+     *
+     * @return a formatted message for an alarm response indicating the time the device rang
+     */
+    public String getAlarmResponseMessage(double time) {
```
Comment (ricdezen) [inline app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java]: I heavily suggest a long value for the time, measured in milliseconds

Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java]: Well, it's not much different, why?

File: .idea/misc.xml (hunk 5 above)
Comment (turcato-niccolo) [reply .idea/misc.xml]: 'honk'

File: app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java (hunk 6 above)
Comment (ricdezen) [reply app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java]: This is due to the fact it becomes easier on both ends to use a `long` value. You can just use `System.currentTimeMillis()` to measure it and `Calendar`+`DateUtils` to format it.

File: app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java [hunk 7]
```diff
@@ -30,4 +34,31 @@ protected static String composeRequestAlarm() {
...
     }
 
+
+    /**
+     * Compose an alarm response message that informs of the amount time the device rang
+     *
+     * @param time The amount time the the device rang (seconds)
+     * @return The correctly composed alarm response
+     */
+    protected static String composeResponseAlarm(double time) {
```
Comment (ricdezen) [inline app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java]: same as above

File: app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java (hunk 6 above)
Comment (turcato-niccolo) [reply app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmManager.java]: Makes perfect sense

**Comment by RicDeZen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5060801&bid=16655812)

**Comment by ricdezen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/ingsw/group1/findmyphone/alarm/AlarmMessageHelper.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5060801&bid=16655812)

**Review by ricdezen — APPROVED** [review]

**Pull request merged**